plugin/storage/grpc/config.go

```diff
@@ -38,6 +38,7 @@ func DefaultConfigV2() ConfigV2 {
 
 func (c *Configuration) TranslateToConfigV2() *ConfigV2 {
 	return &ConfigV2{
+		Tenancy: c.TenancyOpts,
```

Comment:
@mahadzaryab1 I think this is another example of divergence between v1/v2 configs - this is why my preference is always NOT to have separate configs, so that omissions like this won't happen. Do you want to take a look at that as part of #5229?

Reply:
@frzifus what kind of unit or e2e test could we have to catch this bug?

Reply:
I was going to add a unittest for this translate config function. But when I started to re-type this 4 line logic, I thought it doesnt make much sense.

An e2e test that runs jaeger-query with the `SPAN_STORAGE_TYPE=grpc` storage and the `--multi-tenancy.header=x-scope-orgid` flag and checks the metadata of the outgoing gRPC request would be perfect to avoid that issue in the future.
Similar to this: https://github.com/jaegertracing/jaeger/pull/6030#discussion_r1782742118 

@yurishkuro Would it be ok, creating an issue to keep track and submit something afterwards?

Reply:
will do @yurishkuro

Reply:
> An e2e test that runs jaeger-query with the SPAN_STORAGE_TYPE=grpc storage and the --multi-tenancy.header=x-scope-orgid flag and checks the metadata of the outgoing gRPC request would be perfect to avoid that issue in the future.

We have a grpc storage e2e test today. We can add a test there that will run the storage with tenancy enabled and store a span with tenant1 then try to retrieve it with tenant1 (found) and tenant2 (not found). The memory storage used in that e2e test already supports separation of data by tenant.

Reply:
It would be even better to do this with jaeger-v2 e2e tests, but one thing that I don't know about is how we can thread the tenant ID through the OTEL Collector pipeline, from OTLP receiver to jaeger-storage-extension.